chain/network/src/raw/connection.rs

```diff
@@ -69,8 +69,8 @@ impl std::fmt::Debug for Connection {
 
 #[derive(thiserror::Error, Debug)]
 pub enum ConnectError {
-    #[error("IO")]
-    IO(#[source] std::io::Error),
+    #[error(transparent)]
+    IO(std::io::Error),
```

Comment:
"transparent" is fine, but please keep "source" as well. These serve different purposes.

Reply:
not possible actually, you get an error saying: `"transparent error struct can't contain #[source]"`. If I understand correctly though, deleting #[source] and adding transparent will make it so that when you call `source()` on this error type itll just give whatever `source()` returns for the underlying `io::Error`, so that behavior seems ok for what we want I guess

Reply:
ack